### ISSUE TITLE
🚀 Feature: Add type annotations to APIs of the dart_appwrite plugin when using for cloud functions

### ISSUE THREAD

**Issue by arrowxpr** [appwrite/appwrite#7146]
### 🔖 Feature description

The dart runtime in Appwrite's functions is amazing for flutter developers to quickly write and deploy cloud functions written in dart. One major challenge is the lack of type annotations on the APIs. As a flutter developer I am used to strict linting rules from the lint package. It would be great to have types and prevent bugs before they even occur. 

### 🎤 Pitch

Productivity gets plummeted for most flutter/dart developers when working on projects that lack type annotations. Therefore, Adding types to the `dart_appwrite` plugin, when used for cloud functions, will speed up workflows and improve productivity.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@arrowxpr, thanks for creating this issue! 🙏🏼 Are you referring to type information for the context?

**Comment by arrowxpr** [comment]
> @arrowxpr, thanks for creating this issue! 🙏🏼 Are you referring to type information for the context?

Yes. Along with other objects such as req and res as well as their properties. It would be great!
